---
title: "4. Alpha_diversity_analysis"
author: "Baptiste Oosterlinck"
date: "16-2-2022"
output: html_document
---

##1. Summary barplot of the mean abundance (figure 3-A)

```{r}
#absolute abundances of the tumour samples
agglom_BA2_T <- subset_samples(agglom_BA2, Tissue.Type == "tumor") %>%
  tax_glom(.,"Phylum")

#absolute abundances of the tumour samples per mucin phenotype
agglom_BA2_T_Gastric <- subset_samples(agglom_BA2_T, Mucin.Phenotype == "Gastric")
agglom_BA2_T_Intestinal <- subset_samples(agglom_BA2_T, Mucin.Phenotype == "Intestinal")
agglom_BA2_T_Mixed <- subset_samples(agglom_BA2_T, Mucin.Phenotype == "Mixed")
agglom_BA2_T_Null <- subset_samples(agglom_BA2_T, Mucin.Phenotype == "Null")

#absolute abundances for the tumour adjacent tissues and the controls
agglom_BA2_A <- subset_samples(agglom_BA2, Tissue.Type == "adjacent")  %>%
  tax_glom(.,"Phylum") 
agglom_BA2_C <- subset_samples(agglom_BA2, Tissue.Type == "noninflammed") %>%
  tax_glom(.,"Phylum")


#dividing Camiplobacterota into Helicobacter and Non-Helicobacter reads
TempCamp <- subset_taxa(agglom_BA2, Phylum == "Campilobacterota")
TempCamp_Helico <- subset_taxa(TempCamp, Genus == "Helicobacter")
TempCamp_nonHelico <- subset_taxa(TempCamp, Genus != "Helicobacter")

ASV_total_abundance <- data.frame(Tumor = taxa_sums(agglom_BA2_T), 
                                  Gastric = taxa_sums(agglom_BA2_T_Gastric),
                                  Intestinal = taxa_sums(agglom_BA2_T_Intestinal),
                                  Mixed = taxa_sums(agglom_BA2_T_Mixed),
                                  Null = taxa_sums(agglom_BA2_T_Null),
                                  Adjacent =  taxa_sums(agglom_BA2_A), 
                                  Control =  taxa_sums(agglom_BA2_C))

ASV_Camp_abundance <- data.frame(Tumor = c(subset_samples(TempCamp_Helico, Tissue.Type == "tumor") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums(),
                                   subset_samples(TempCamp_nonHelico, Tissue.Type == "tumor") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()%>%
                                   sum())
                                 ,Gastric = c(subset_samples(TempCamp_Helico, Tissue.Type == "tumor" & Mucin.Phenotype == "Gastric") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()
                                 ,subset_samples(TempCamp_nonHelico, Tissue.Type == "tumor" & Mucin.Phenotype == "Gastric") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums() %>%
                                   sum())
                                 ,Intestinal = c(subset_samples(TempCamp_Helico, Tissue.Type == "tumor" & Mucin.Phenotype == "Intestinal") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()
                                 , subset_samples(TempCamp_nonHelico, Tissue.Type == "tumor" & Mucin.Phenotype == "Intestinal") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums() %>%
                                   sum())
                                 ,Mixed = c(subset_samples(TempCamp_Helico, Tissue.Type == "tumor" & Mucin.Phenotype == "Mixed") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()
                                 , subset_samples(TempCamp_nonHelico, Tissue.Type == "tumor" & Mucin.Phenotype == "Mixed") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums() %>%
                                   sum())
                                 ,Null = c(subset_samples(TempCamp_Helico, Tissue.Type == "tumor" & Mucin.Phenotype == "Null") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()
                                 , subset_samples(TempCamp_nonHelico, Tissue.Type == "tumor" & Mucin.Phenotype == "Null") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums() %>%
                                   sum())
                                 ,Adjacent = c(subset_samples(TempCamp_Helico, Tissue.Type == "adjacent") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()
                                 , subset_samples(TempCamp_nonHelico, Tissue.Type == "adjacent") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums() %>%
                                   sum())
                                 ,Control = c(subset_samples(TempCamp_Helico, Tissue.Type == "noninflammed") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()
                                 , subset_samples(TempCamp_nonHelico, Tissue.Type == "noninflammed") %>%
                                   tax_glom(.,"Genus") %>%
                                   taxa_sums()%>%
                                   sum()))

ASV_total_relAbundance <- apply(ASV_total_abundance, MARGIN = 2, function(x) (x/sum(x))*100) %>%
  as.data.frame() %>%
  mutate(.,ASV = row.names(.))

ASV_total_abundance <- ASV_total_abundance %>%
  as.data.frame() %>%
  mutate(.,ASV = row.names(.))

ASV_taxonomy <- tax_table(agglom_BA2_T) %>%
  as.data.frame(.)%>%
  mutate(., ASV = row.names(.))

ASV_total <- apply(ASV_total_abundance, MARGIN = 2, function(x) sum(x))
ASV_Camp_relAbundance <- t(apply(ASV_Camp_abundance, MARGIN = 1, function(x) 100 * x / ASV_total))

ASV_total_relAbundance <- full_join(ASV_taxonomy, ASV_total_relAbundance, by = "ASV")
ASV_total_abundance <- full_join(ASV_taxonomy, ASV_total_abundance, by = "ASV")

ASV_total_relAbundance <- ASV_total_relAbundance[,c("Phylum","Control","Adjacent","Tumor","Gastric","Intestinal","Mixed","Null")]

rownames(ASV_total_relAbundance) <- ASV_total_relAbundance$Phylum

chitest_overall <- chisq.test(x= ASV_total_relAbundance[,c("Control","Adjacent","Tumor")])
chitest_camp_O <- chisq.test(x = ASV_Camp_relAbundance[2, c("Control","Adjacent","Tumor")])
chitest_camp_m <- chisq.test(x = ASV_Camp_relAbundance[2, c("Control","Gastric","Intestinal","Mixed","Null")])

print(chitest_camp_O)
print(chitest_camp_m)

ASV_total_relAbundance$Chi2_overall <- NA
ASV_total_relAbundance$Chi2_pheno <- NA

for (phy in ASV_total_relAbundance$Phylum){
  ASV_temp_overall <- ASV_total_relAbundance[ASV_total_relAbundance$Phylum == phy,c("Control", "Adjacent","Tumor")]
  ASV_temp_tumor <- ASV_total_relAbundance[ASV_total_relAbundance$Phylum == phy,c("Control","Gastric","Intestinal","Mixed","Null")]
  ASV_total_relAbundance$Chi2_overall[ASV_total_relAbundance$Phylum == phy] <- chisq.test(ASV_temp_overall)$p.value
  ASV_total_relAbundance$Chi2_pheno[ASV_total_relAbundance$Phylum == phy] <- chisq.test(ASV_temp_tumor)$p.value
}

writexl::write_xlsx(ASV_total_relAbundance, path = "ProportionalAnalysis_Phylum.xlsx")
rm(ASV_total_abundance,nonPaired.PID,phy)

ASV_total_relAbundance_long <- gather(data = ASV_total_relAbundance, key = Tissue.Type, value = Relative.Abundance, 2:8)
BarSummary_Phylum <- ggplot(data =  ASV_total_relAbundance_long, aes(x = Tissue.Type, y = Relative.Abundance, fill = Phylum))+
  geom_bar(stat = "identity", color = "black")+
  scale_x_discrete(guide = guide_axis(angle = 90))+
  ylab("Relative abundance")+
  xlab("")+
  guides(fill = guide_legend(ncol = 1))+
  scale_fill_viridis(option = "turbo",discrete = TRUE)+
  theme_classic2()
BarSummary_Phylum

rm(ASV_taxonomy,ASV_temp_overall,ASV_temp_tumor,ASV_total_relAbundance_long,agglom_BA2_A,agglom_BA2_C,agglom_BA2_T,agglom_BA2_T_Gastric,agglom_BA2_T_Intestinal,agglom_BA2_T_Mixed,agglom_BA2_T_Null)
```

##2. calculating the diversity measures we are interested in:
  - Shanon
  - Chao1
  - Simpson
  - Richness
  
```{r}
#calculating the indexes of interest
Alpha.Div <- microbiome::alpha(agglom_relAb2,index = c("observed","diversity_shannon","chao1","diversity_inverse_simpson"))
Alpha.Div$Illumina.ID <- row.names(Alpha.Div)

#getting the sample data from the phyloseq object and merging it with the alpha diversity measures
AlphaDiv_DF <- sample_data(agglom_relAb2) %>%
  data.frame(.) %>%
  merge(.,Alpha.Div, by = "Illumina.ID")
```

##3. plotting the different diversity measures according to mucin phenotype (figure 3-B)

```{r}
#creating long format for the mucin data
AlphaDiv_long <- gather(data = AlphaDiv_DF , key = "Index", value = "Value",  67:70)
AlphaDiv_long$Mucin.Phenotype <- factor(AlphaDiv_long$Mucin.Phenotype, levels = c("control","adjacent","tumor","Gastric","Intestinal","Mixed","Null"),labels =c("control","adjacent","tumor","Gastric","Intestinal","Mixed","Null"))
AlphaDiv_long[AlphaDiv_long$Tissue.Type == "noninflammed","Mucin.Phenotype"] <- "control"
AlphaDiv_long[AlphaDiv_long$Tissue.Type == "adjacent","Mucin.Phenotype"] <- "adjacent"

AlphaDiv_extended <- AlphaDiv_long %>%
  filter(.,Tissue.Type == "tumor") %>%
  mutate(.,Mucin.Phenotype = .$Tissue.Type) %>%
  rbind(.,AlphaDiv_long)%>%
  drop_na(.,Mucin.Phenotype)

AlphaDiv_extended$Mucin.Phenotype <- droplevels(AlphaDiv_extended$Mucin.Phenotype) %>%
  factor(.,levels = c("control","adjacent","tumor","Gastric","Intestinal","Mixed","Null"),labels =c("Control","Adjacent","Tumor","Gastric","Intestinal","Mixed","Null"),ordered = TRUE)

#generating general plot for the diversity indexes
ALphaDiversity_boxplot <- AlphaDiv_extended %>%
  ggplot(data = ., aes(x = Mucin.Phenotype , y = Value, color = Mucin.Phenotype))+
  geom_boxplot()+
  geom_point(position = position_jitter(seed = 1995,width = 0.15))+
  facet_wrap(~Index,scales = "free_y")+
  scale_x_discrete(guide = guide_axis(angle = 90))+
  stat_compare_means(aes(label = ..p.signif..),hide.ns = TRUE,method = "wilcox.test",comparisons = list(c("Gastric","Intestinal"),c("Intestinal","Mixed"),c("Mixed","Null"),c("Gastric","Mixed"),c("Gastric","Null"),c("Intestinal","Null")))+
  theme_classic2()
ALphaDiversity_boxplot
```
